kmip/services/server/engine.py

```diff
@@ -1652,10 +1652,22 @@ def _process_mac(self, payload):
         if payload.data:
             data = payload.data.value
         else:
-            raise exceptions.InvalidField(
+            raise exceptions.PermissionDenied(
                 "No data to be MACed"
             )
 
+        if managed_object.state != enums.State.ACTIVE:
+            raise exceptions.PermissionDenied(
+                "Object is not in a state that can be used for MACing."
+            )
+
+        if enums.CryptographicUsageMask.MAC_GENERATE not in \
+           managed_object.cryptographic_usage_masks:
+                raise exceptions.PermissionDenied(
+                    "MAC GERERATE crypto usage mask must be set "
+                    "for the MAC operation"
```

Comment:
Let's change the error message to: "MAC Generate must be set in the object's cryptographic usage mask"